### PR TITLE
#187 Create new accordion variant

### PR DESCRIPTION
**Pull request by katarinazaprazna**
- Add new variant `waste`
Screenshot 2024-07-23 at 06 14 29

### CODE REVIEW

File: next/src/components/common/Accordion/Accordion.tsx [hunk 1]
```diff
@@ -18,17 +21,36 @@ export type AccordionProps = {
...
+        <details className={cn('group flex w-full flex-col', className)}>
           <summary
             className={cn(
-              'flex cursor-pointer items-center gap-4 text-left after:absolute after:inset-0',
-              'ring-offset-2 focus:outline-none focus-visible:ring',
+              'relative flex cursor-pointer items-center justify-center gap-4 text-left after:absolute after:inset-0',
+              'ring-offset-2 group-open:pb-0 focus:outline-none focus:ring',
+              {
+                'py-5 group-open:border-none group-open:py-5 group-open:pb-3 lg:group-open:pb-4':
+                  icon === undefined,
```
Comment (radoslavzeman) [inline next/src/components/common/Accordion/Accordion.tsx:42]: This condition will catch only `undefined`, but `icon` can also be `null` as it is typed as `ReactNode` https://reactnative.dev/docs/react-node.
It's completely okay to write simple `'some-tailwind-clases-here': !icon` instead.

File: next/src/components/common/Accordion/Accordion.tsx [hunk 2]
```diff
@@ -18,17 +21,36 @@ export type AccordionProps = {
...
-      <div className={cn('bg-background-primary', className)}>
-        <details className={cn('group flex w-full flex-col py-5', {})}>
+      <div className="bg-background-primary">
+        <details className={cn('group flex w-full flex-col', className)}>
           <summary
             className={cn(
-              'flex cursor-pointer items-center gap-4 text-left after:absolute after:inset-0',
-              'ring-offset-2 focus:outline-none focus-visible:ring',
+              'relative flex cursor-pointer items-center justify-center gap-4 text-left after:absolute after:inset-0',
+              'ring-offset-2 group-open:pb-0 focus:outline-none focus:ring',
```
[the remark this answers is not in the document]
Comment (radoslavzeman) [reply next/src/components/common/Accordion/Accordion.tsx:39]: We can maybe fix focus rings in a separate PR as it's a kind of complex topic? @Ty-ci

File: next/src/components/common/Accordion/Accordion.tsx
Comment (radoslavzeman) [inline next/src/components/common/Accordion/Accordion.tsx:1]: It seems redundant to have `border-b border-border-default` on two places, once controlled by `hasBottomBorder` and once controlled by additional `innerClassName`.

**Review by radoslavzeman — COMMENTED** [review]
I think we are very close to merge 💯 😄 
Most of the comments are not critical, I see very nice progress. I also see that there is still some confusion how to use dividers and I guess it's because of Figma design, and that's okay.
Even though comments are not critical, we still want to maintain clean code standards and comments from both @Ty-ci and me should be addressed, but the Menu has higher priority than this PR.

File: next/src/components/common/Accordion/WasteIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+const WasteIcon = ({
+  variant,
+}: {
+  variant: 'paper' | 'plastic' | 'glass' | 'mixed' | 'organic'
+}) => {
+  return (
+    <div
+      className={cn('flex h-[3rem] w-[3rem] rounded-2xl p-3 text-white', {
+        'bg-waste-paper': variant === 'paper',
+        'text-black bg-waste-plastic': variant === 'plastic',
```
[the remark this answers is not in the document]
Comment (katarinazaprazna) [reply next/src/components/common/Accordion/WasteIcon.tsx:19]: These are the inconsistencies I asked you about yesterday. `text-white` should also be removed

File: next/src/components/common/Accordion/Accordion.tsx
Comment (katarinazaprazna) [reply next/src/components/common/Accordion/Accordion.tsx:1]: It was difficult to handle two different Accordion displays within one component. Each has different behaviour for border display and how many borders should be visible. This led to code duplication - I can check again whether some of them can be indeed removed

File: next/src/components/common/Accordion/WasteSortingGuide.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+
+import Icon from '@/src/components/common/Icon/Icon'
+import Typography from '@/src/components/common/Typography/Typography'
+import cn from '@/src/utils/cn'
+
+type WasteSortingColumnProps = {
+  title: string
+  items: { title: string }[]
```
[the remark this answers is not in the document]
Comment (katarinazaprazna) [reply next/src/components/common/Accordion/WasteSortingGuide.tsx:9]: That's a good observation. However, if we remove `item: any` type from the loop, the simplified code gives us error about `title` not being defined on the type. If removed and kept as such `items: { title: string }[]`, Webstorm does not give me any error

File: next/src/components/common/Accordion/Accordion.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (katarinazaprazna) [reply next/src/components/common/Accordion/Accordion.tsx:39]: @Ty-ci After implementing the focus-ring related code, I see no difference with the ring display on Safari and Arc. We can discuss this on Monday. I also found a Safari-related bug. In Safari, there's a carret displayed for expandable blocks
Screenshot 2024-07-27 at 16 08 40

[the remark this answers is not in the document]
Comment (radoslavzeman) [reply next/src/components/common/Accordion/Accordion.tsx:42]: Nice catch!
We solved this in bratislava.sk here https://github.com/bratislava/bratislava.sk/blob/master/next/styles/globals.css#L230

**Pull request merged**